reference_data/management/tests/update_gencode_tests.py

```diff
@@ -0,0 +1,207 @@
+
+        transcript_infos = {trans.transcript_id: {
+            'start_grch37': trans.start_grch37,
+            'end_grch37': trans.end_grch37,
+            'strand_grch37': trans.strand_grch37,
+            'chrom_grch37': trans.chrom_grch37,
+            'gene_id': trans.gene.gene_id
+        } for trans in TranscriptInfo.objects.all()}
+        self.assertEqual(len(transcript_infos), 2)
+        self.assertDictEqual(transcript_infos['ENST00000456328'], {'start_grch37': 11869, 'end_grch37': 14409, 'strand_grch37': u'+', 'chrom_grch37': u'1', 'gene_id': u'ENSG00000223972'})
+        self.assertDictEqual(transcript_infos['ENST00000332831'], {'start_grch37': 621059, 'end_grch37': 622053, 'strand_grch37': u'-', 'chrom_grch37': u'1', 'gene_id': u'ENSG00000284662'})
+
+        # Test normal command function without a --reset option
+        mock_logger.reset_mock()
+        call_command('update_gencode', '--reset', '--gencode-release=31', self.temp_file_path, '37')
+        calls = [
+            mock.call('Dropping the 2 existing TranscriptInfo entries'),
+            mock.call('Dropping the 50 existing GeneInfo entries'),
+            mock.call(
+                'Loading {} (genome version: 37)'.format(self.temp_file_path)),
+            mock.call('Creating 2 GeneInfo records'),
+            mock.call('Creating 2 TranscriptInfo records'),
+            mock.call('Done'),
+            mock.call('Stats: '),
+            mock.call('  genes_created: 2'),
+            mock.call('  transcripts_created: 2')
+        ]
+        mock_logger.info.assert_has_calls(calls)
+
+        gene_infos = {gene.gene_id: {
+            'start_grch37': gene.start_grch37,
+            'chrom_grch37': gene.chrom_grch37,
+            'coding_region_size_grch37': gene.coding_region_size_grch37,
+            'gencode_release': gene.gencode_release,
+            'gencode_gene_type': gene.gencode_gene_type,
+            'gene_id': gene.gene_id,
+            'gene_symbol': gene.gene_symbol,
+            'end_grch37': gene.end_grch37,
+            'strand_grch37': gene.strand_grch37
+        } for gene in GeneInfo.objects.all()}
+        self.assertEqual(len(gene_infos), 2)
+        self.assertDictEqual(gene_infos['ENSG00000223972'], {'start_grch37': 11869, 'chrom_grch37': u'1', 'coding_region_size_grch37': 0, 'gencode_release': 31, 'gencode_gene_type': u'transcribed_unprocessed_pseudogene', 'gene_id': u'ENSG00000223972', 'gene_symbol': u'DDX11L1', 'end_grch37': 14409, 'strand_grch37': u'+'})
+        self.assertDictEqual(gene_infos['ENSG00000284662'], {'start_grch37': 621059, 'chrom_grch37': u'1', 'coding_region_size_grch37': 936, 'gencode_release': 31, 'gencode_gene_type': u'protein_coding', 'gene_id': u'ENSG00000284662', 'gene_symbol': u'OR4F16', 'end_grch37': 622053, 'strand_grch37': u'-'})
```

Comment:
As above, test line by line and only test all the fields for one entry

Reply:
As above.